pkg/kv/kvnemesis/watcher.go

```diff
@@ -192,6 +193,44 @@ func (w *Watcher) processEvents(ctx context.Context, eventC chan kvcoord.RangeFe
 					}
 				}
 				w.mu.Unlock()
+			case *roachpb.RangeFeedSSTable:
+				w.mu.Lock()
+				if len(e.Data) == 0 {
+					return errors.AssertionFailedf("no SST data found")
```

Comment:
The locking here is broken, lots of error returns that never unlock.

What I don't understand is why the other code branches do this too, is this ok for some reason?

Reply:
Could you pull this out into `w.handleAddSSTable`? That will solve your locking problem and also declutter this method.

If you feel generous you could do the same for the rangefeed checkpoint case above.

Reply:
The test will fail and exit on any errors, so it doesn't really matter that we're holding the lock, and I figured since we already did this for other branches it was "fine". Anyway, will pull them out into separate methods.